example/stochastic_subgrid/stochastic_subgrid_package.cpp

```diff
@@ -291,10 +291,10 @@ void DoLotsOfWork(MeshBlockData<Real> *rc) {
   // packing in principle unnecessary/convoluted here and just done for demonstration
   PackIndexMap imap;
   std::vector<std::string> vars({"num_iter", "advected", "dummy_result"});
-  auto v = rc->PackVariables(vars, imap);
-  const int niter = imap["num_iter"].first;
-  const int in = imap["advected"].first;
-  const int out = imap["dummy_result"].first;
+  auto v = rc->PackVariables(vars, false, &imap);
+  const int niter = imap.get("num_iter").first;
+  const int in = imap.get("advected").first;
+  const int out = imap.get("dummy_result").first;
```

Comment:
I got burnt by `[]` not telling me that I'm requesting a non-existent field. If you add a sparse pool called "sparse" with sparse ID, say, 0, then you can only get it with `imap.get("sparse_0")` or `imap.get("sparse", 0)`. I got burnt by `imap.get("sparse")` returning something (something nonsensical, but without throwing) and then trying to use that.

I still don't understand the use case for having `[]` return {0,-1} if you request a non-existent variable. If one just blindly calls `first` (which will be 0) and uses that index, one will be accessing another variable (whichever happened to be first in the pack) instead of the one that was intended to be accessed. That sounds like a recipe for hard-to-find bugs.

To check if a variable exists in the pack, one can use the `PackIndexMap::Has` function. So before one would get an index pair and check if it's valid, now one checks if the pack has the variable and then gets the (valid) index pair.

Furthermore, `operator[]` can only take a single argument, so we couldn't use it to get a sparse field with base name and sparse ID, we'd be forced to use the full label.

Reply:
> If one just blindly calls first (which will be 0) and uses that index, one will be accessing another variable (whichever happened to be first in the pack) instead of the one that was intended to be accessed. That sounds like a recipe for hard-to-find bugs.

This is true. I would be open to making first and second both -1 for nonexistent variables.

> To check if a variable exists in the pack, one can use the PackIndexMap::Has function. So before one would get an index pair and check if it's valid, now one checks if the pack has the variable and then gets the (valid) index pair.

Ah okay. I missed that we have the ability to check. That's probably sufficient. That said, this is very API breaking. We use the `[]` everywhere throughout the downstream codes. Can we provide both `[]` and `get`? Then the user can decide to use the old `[]` accessor if they have good reason to. This would be equivalent to the way `std::map` works.

Reply:
I am still exposing `[]` but it just forwards to `get` and would throw if the variable doesn't exist. I also added deprecated warning to `[]`, but I could remove that. Is this what you would like, or would you like `[]` to have the old behavior and not throw?

Of course, we can't provide `[]` for sparse base name and id, one would have to combine them together into a single label string.

Reply:
Can we just let `[]` not throw and have the old behaviour? I agree the new API is nice, but we're already using the old API, with checks for negative `map.second` in a bunch of places.

Reply:
Awesome. Thanks.